src/transforms/pipelines/config.rs

```diff
@@ -42,62 +52,178 @@ impl Clone for PipelineConfig {
     }
 }
 
+#[async_trait::async_trait]
+#[typetag::serde(name = "pipeline")]
+impl TransformConfig for PipelineConfig {
+    async fn build(&self, ctx: &TransformContext) -> crate::Result<Transform> {
+        let condition = match &self.filter {
+            Some(config) => Some(config.build(&ctx.enrichment_tables)?),
+            None => None,
+        };
+
+        // Setup the interior transform chain in this pipeline.
+        //
+        // NOTE: As this pipeline is not expanding to separate transforms we
+        // lose validation that happens at the topology layer (e.g. making sure
+        // input/output types of connected transforms are compatible). This is
+        // an unfortunate consequence of having a "topology in a topology".
+        //
+        // In the future, it would be great to decouple that type of validation
+        // from the physical task boundaries. You could imagine a future where
+        // we replace transform expansion with the ability to return different
+        // types of compound transforms directly from TransformConfig::build,
+        // and those know how to play nice with a subsequent graph-based
+        // validation scheme.
+        //
+        // For now, considering that most everything is log to log with regard
+        // to pipelines we avoid adding duplicate validation here in favor of
+        // future work.
+        if self.transforms.is_empty() {
+            return Err(format!("empty pipeline: {}", self.name).into());
+        }
+        // Today we make the assumption that to be a valid pipeline transform
+        // the transform CANNOT have named outputs. This assumption might break
+        // in the future so, to avoid panics, we instead make building a
+        // pipeline with such transforms an error.
+        for transform in &self.transforms {
+            if transform.outputs(&ctx.merged_schema_definition).len() > 1 {
+                return Err(format!(
+                    "pipeline {} has transform of type {} with a named output, unsupported",
+                    self.name,
+                    transform.transform_type()
+                )
+                .into());
+            }
```

Comment:
In this case it's my understanding that we're just replicating the existing model on the Datadog Logs side. It's not hugely more complicated to handle multiple outputs, but at this point it does seem like a translation bug if they're present.

Reply:
Yeah, agreed for automatic translation, but as mentioned in my other comment, this will be a feature generally available to all users as a way to organize topology definitions so I expect we will see users handwriting pipeline configurations.

Reply:
This one is worth a follow-up issue I think and is a blocker to the pipeline being generally available to end users. In-house it's okay for the undocumented feature to be a little sharp-edged.

Reply:
I agree it seems like we need a follow up issue to discuss what these transforms should look like as a more generally available feature. They're currently built to match a more restrictive model, and we can't just remove those restrictions and have the resulting system make sense.